### PR TITLE
build: make sure eslint-plugin-calcite-components is set up correctly

### PR DESCRIPTION
**Pull request by benelan**
**Related Issue:** #

## Summary

I forgot to mention that the packages within the monorepo are "local dependencies", which means `npm install` won't install them, they have to be built locally. A common workflow is `cd`ing directly into calcite-components to do work. This caused issues with `eslint-plugin-calcite-components` because it wasn't being built so it wasn't working during development in CC. 

I added a `start` turbo pipeline which will ensure all dependencies are built first. So running `npm start` from the root directory will no longer cause this issue.

I also cleaned up the Contributing doc and tweaked some eslint settings that were causing issues in VSCode.

### CODE REVIEW

**Review by driskull — APPROVED** [review]
Awesome 👏

**Comment by benelan** [comment]
@jcfranco I had to remove the on-exit cleanup of js demo files from `npm start` because it was breaking the turbo pipeline. I'm guessing turbo's concurrency was getting tangled up `concurrently` and the exit code watcher. 

Instead, I clean out all of the js files created by `tsc` before running `npm start` and make sure the files that were previously being cleaned are gitignored. 

Is that solution okay with you or should I spend more time troubleshooting the concurrency issue?

File: CONTRIBUTING.md
Comment (jcfranco) [inline CONTRIBUTING.md:115]: Nitpick: I feel like we need an `"and then"` between cloning and installing deps. I'll leave this up to you.

File: packages/eslint-plugin-calcite-components/.eslintrc.json
Comment (jcfranco) [inline packages/eslint-plugin-calcite-components/.eslintrc.json:2]: Any package with this configuration will skip linting entirely, right? If so, won't this make the contributing instructions on linting misleading?

Comment (jcfranco) [reply packages/eslint-plugin-calcite-components/.eslintrc.json:2]: For now, maybe the contributing doc can be updated to indicate that ESLint will only run on `calcite-components` and we can tweak our linting config to lint properly outside `calcite-components` after this lands?

**Review by jcfranco — APPROVED** [review]
Aside from a comment regarding linting on other packages, this LGTM! 🚀

File: packages/eslint-plugin-calcite-components/.eslintrc.json
Comment (benelan) [reply packages/eslint-plugin-calcite-components/.eslintrc.json:2]: Yeah I added this to CCR and the eslint plugin so eslint won't complain about there not being a config which covers their files.  

I'm working on pulling shared eslint/tsconfigs into separate workspaces as part of the lint-staged fix. After that happens, the base eslint config will be used in these packages. 

So I'm going to leave the doc as is for now so I don't have to change it and then change it back.

**Comment by benelan** [comment]
Note: I couldn't get the tsconfig/eslint separating to work at the moment so I updated the doc. See https://github.com/Esri/calcite-design-system/issues/7122#issuecomment-1626229399

**Pull request merged**
